src/client/chat.js

```diff
@@ -222,7 +222,7 @@ module.exports = function (client, options) {
       })
 
       if (packet.signature) {
-        client._lastSeenMessages.push(packet.signature)
+        client._lastSeenMessages.push({ signature: packet.signature, pending: false })
```

Comment:
This won't work as signature needs to be a Buffer. The push method on lastSeenMessages also handles pending state internally already.

Furthermore, signatures must be pending: true by default or chat signing will break after a hide_chat packet

Reply:
The LastSeenMessages object was sending signatures for messages with pending set to false. Is that what it was supposed to be doing? If you set pending to true here, it sends the signature containing only msgs it hasn't sent before. However, in both of these cases, I get kicked from the vanilla server when I send commands. If I set pending to false, then i never get kicked from offline or online vanilla servers, but that's because it doesn't send signatures, so I'm not sure what else that breaks. Do you know of things I should test to make sure that this doesn't break anything that is currently working?

As for hidden messages, those are handled by setting the whole entry to null, and those don't use pending as far as I can tell. That logic should still be the same.

Reply:
Wait, I found what I did wrong. Chat messages set pending to 0 afterwards, I forgot that for commands. Adding that makes everything work. I just pushed that version